pkgs/build-support/build-graalvm-native-image/default.nix

```diff
@@ -16,14 +16,13 @@
     "-H:Name=${executable}"
     "-march=compatibility"
     "--verbose"
-    "-J-Dsun.stdout.encoding=UTF-8"
-    "-J-Dsun.stderr.encoding=UTF-8"
   ]
   # Extra arguments to be passed to the native-image
 , extraNativeImageBuildArgs ? [ ]
   # XMX size of GraalVM during build
 , graalvmXmx ? "-J-Xmx6g"
 , meta ? { }
+, LC_ALL ? "en_US.UTF-8"
```

Comment:
is it possible to use C.UTF-8 ?

Reply:
You can always override it yourself.

Reply:
the intention of the question was not to assume en_US as default for users, if the issue is fixed with C.UTF-8 then en_US shouldn't be default

Reply:
AFAIK I remember C.UTF-8 causes some weird issues. You can always use it yourself, but I don't think it should be the default.

Reply:
Also doesn't work:

```
/nix/store/0wzb184kckhmggq1nlvy5xk6rykp0jl1-bash-5.2-p21/bin/sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8): No such file or directory
```

Reply:
```
 git diff
diff --git a/pkgs/build-support/build-graalvm-native-image/default.nix b/pkgs/build-support/build-graalvm-native-image/default.nix
index f0e73901c8db..d19eeb915914 100644
--- a/pkgs/build-support/build-graalvm-native-image/default.nix
+++ b/pkgs/build-support/build-graalvm-native-image/default.nix
@@ -22,7 +22,7 @@
   # XMX size of GraalVM during build
 , graalvmXmx ? "-J-Xmx6g"
 , meta ? { }
-, LC_ALL ? "en_US.UTF-8"
+, LC_ALL ? "C.UTF-8"
 , ...
 } @ args:
 

[nix-shell:~/.cache/nixpkgs-review/pr-283973/nixpkgs]$ nix-build -A babashka
/nix/store/hrp07d18wnfhb7kvd7w5hyni9mb2nmsm-babashka-1.3.188
```

Reply:
Are you running this in Linux? I tried this and doesn't work in `aarch64-darwin`.

Reply:
yes, this is on x86_64-linux

also, can you point me to why `env.` is persisted beyond builds, I thought it was only env for builder

Reply:
Because GraalVM captures those environment variables at build time: https://github.com/oracle/graal/pull/6095

Reply:
> yes, this is on x86_64-linux

So the weird issue with `C.UTF-8` that I remember is basically that, it works in Linux and doesn't work in Darwin.

I much prefer to be sane here and default to `en_US.UTF-8` so at least this is consistent between different platforms. Anyone that needs `C.UTF-8` in Linux can override.

Reply:
thank you, some sanity restored

I think en_US is fine, my question was mainly not to assume all UTF is en_US